src/main/java/org/folio/inventory/resources/Instances.java

```diff
@@ -326,57 +336,28 @@ private void getById(RoutingContext routingContext) {
    *
    * @param instancesResponse Multi record Instances result
    * @param routingContext
-   * @param context
    */
-  private CompletableFuture<InstancesResponse> fetchRelationships(
+  private Future<InstancesResponse> fetchRelationships(
```

Comment:
Why has this code switched from using `CompletableFuture` to Vert.x `Future`?

Reply:
The reason is to be able to propagate failure occurred during fetching relationships/preceding-succeeding titles to client. I'm not sure if it is possible with `CompletableFuture`.

Reply:
What do you mean by propagate failure? Do you mean allow the Vert.x error handler to pick up exception?

Reply:
I have reverted it to use `CompletableFuture` it also supports the error propagation.

Yes, I mean to not execute all further calls in the chain if on some step an exception has been raised and also have a common method which will be executed regardless failure or success, to generate response to client.

Reply:
> I have reverted it to use `CompletableFuture` it also supports the error propagation.

Thank you.

> Yes, I mean to not execute all further calls in the chain if on some step an exception has been raised and also have a common method which will be executed regardless failure or success, to generate response to client.

Yes, if you haven't already (I haven't reviewed the changes yet), please add an `exceptionally` on to the end of the chain to forward any uncaught exception.